dts/bindings/dma/st,stm32-dma.yaml

```diff
@@ -23,15 +23,67 @@ properties:
 
 # Parameter syntax of stm32 follows the dma client dts syntax
 # in the Linux kernel declared in
-# https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/plain/Bindings/dma/stm32-dma.txt
+# https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/plain/Bindings/dma/st,stm32-dma.yaml
 #
-# channel: DMA channel
-# slot: DMA stream of the DMA channel for DMA V1 or for DMA V2 with MUX peripheral request else NA
-# channel-config: configuration of the selected DMA channel
-# features: fifo threshold if relevant for DMA V1 or TBD for DMA V2
 
 dma-cells:
   - channel
   - slot
   - channel-config
   - features
+
+description: |
+  The STM32 DMA is a general-purpose direct memory access controller
+  capable of supporting 5 or 6 or 7 or 8 independent DMA channels.
+  Each channel can have up to 8 requests.
+  DMA clients connected to the STM32 DMA controller must use the format
+  described in the dma.txt file, using a four-cell specifier for each
+  channel: a phandle to the DMA controller plus the following four integer cells:
+    1. channel: the dma stream from 0 in V1 or 1 in V2 to <dma-requests>
+    2. slot: DMA periph request ID in DMA_V2 or DMAMUX, dma request in V1
+    3. channel-config: A 32bit mask specifying the DMA channel configuration
+    which is device dependent:
+        -bit 6-7 : Direction  (see dma.h)
+       	       0x0: MEM to MEM
+       	       0x1: MEM to PERIPH
+       	       0x2: PERIPH to MEM
+       	       0x3: reserved for PERIPH to PERIPH
+        -bit 9 : Peripheral Increment Address
+       	       0x0: no address increment between transfers
+       	       0x1: increment address between transfers
+        -bit 10 : Memory Increment Address
+       	       0x0: no address increment between transfers
+       	       0x1: increment address between transfers
+        -bit 11-12 : Peripheral data size
+       	       0x0: Byte (8 bits)
+       	       0x1: Half-word (16 bits)
+       	       0x2: Word (32 bits)
+       	       0x3: reserved
+        -bit 13-14 : Memory data size
+       	       0x0: Byte (8 bits)
+       	       0x1: Half-word (16 bits)
+       	       0x2: Word (32 bits)
+       	       0x3: reserved
+        -bit 15: Peripheral Increment Offset Size (not USED for DMA V2)
+       	       0x0: offset size is linked to the peripheral bus width
+       	       0x1: offset size is fixed to 4 (32-bit alignment)
+        -bit 16-17 : Priority level
+       	       0x0: low
+       	       0x1: medium
+       	       0x2: high
+       	       0x3: very high
+    4. features: A 32bit bitfield value specifying DMA features
+        -bit 0-1 or not used: DMA FIFO threshold selection
```

Comment:
Not sure I follow here.  What does 'or not used' mean?

Reply:
Depending on the IP, it might not be used (this is the case for _v1)

Reply:
This was actually not used in V2 case. Added a clarification on that.